plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2073,6 +2073,7 @@ void producer_plugin_impl::produce_block() {
    if ( blockvault_plug->get() != nullptr ) {
       std::promise<bool> p;
       std::future<bool> f = p.get_future();
+      consider_new_watermark( pending_blk_state->header.producer, pending_blk_state->block_num, pending_blk_state->block->timestamp );
       std::optional<producer_watermark> watermark{get_watermark(pending_blk_state->header.producer)};
       EOS_ASSERT(watermark.has_value(), empty_watermark, "Attempting to use a watermark that does not exist");
       blockvault_plug->get()->async_propose_constructed_block(watermark.value(), pending_blk_state->dpos_irreversible_blocknum, pending_blk_state->block, [&p](bool b) {
```

Comment:
Remove the calls to `consider_new_watermark` and `get_watermark`. Just do the following:
`blockvault_plug->get()->async_propose_constructed_block({pending_blk_state->block->block_num, pending_blk_state->block->timestamp}, .. `

Reply:
Made requested change.

Reply:
Please update the PR description.

Reply:
The PR description has been updated.